### PR TITLE
perf(cli): trace where a CLI run spends its time

### PR DESCRIPTION
**Pull request by adrians5j**
## What

`webiny --version` takes about four seconds, and there was no way to see what it was doing. This adds a phase tracer and wires it through both CLI entrypoints, the project SDK bootstrap, config rendering and extension validation.

Nothing changes unless you ask for it. This is instrumentation only — no behaviour change, no perf fix. The fixes it points at come next.

## Using it

```bash
yarn webiny <command> --trace       # also --trace=true, or WEBINY_CLI_TRACE=1
```

The report goes to stderr on exit, so it can't corrupt the stdout of a piped command. Phases nest by containment, and anything under a millisecond is folded into a count at the bottom. The config render runs in a forked child process, so it gets a report of its own that the parent forwards.

```
Config render worker trace — 1845 ms total

   856 ms    46.4%   load modules
   789 ms    42.7%   import webiny.config
    81 ms     4.4%   create JSDOM
   117 ms     6.3%   render config tree

Webiny CLI (aws) trace — 5519 ms total

   861 ms    15.6%   load modules
   799 ms    14.5%   check system requirements
   557 ms    10.1%   check @webiny package versions
  3290 ms    59.6%   initialize CLI
  3285 ms    59.5%     initialize project SDK
  1869 ms    33.9%       get project extensions
  1863 ms    33.8%         render project config
   808 ms    14.6%       validate project extensions
   768 ms    13.9%         validate "Admin/AfterDeploy"
   596 ms    10.8%       register project SDK extensions
    11 ms     0.2%   run command (still running at exit)
```

`--trace` is registered as a global option so it shows up in `webiny --help`, but `startTrace` reads it off `process.argv` by hand, because tracing has to be on before yargs is built.

## What it found

Four things, none of them the command being run:

1. **Two preflight checks shell out on every invocation** (~1.4s). `ensureSystemRequirements` makes three blocking `execaSync` calls; `ensureSameWebinyPackageVersions` runs `yarn info "@webiny/*" --all`. A CPU profile shows 1392 ms of `spawnSync` plus 2816 ms parked idle on those pipes.
2. **The config render forks a second Node process** (~2.2s). Half of that is the worker loading its own import graph, another 888 ms is importing `webiny.config.tsx`. The actual React render is 115 ms.
3. **Extension validation imports every extension module** (~750 ms for one extension). `zodSrcPath` dynamically imports each extension's `src` purely to check the exported class implements the right abstraction, and `register project SDK extensions` then imports the same modules again.
4. **Static imports load everything regardless of command** (~750 ms). `@pulumi/aws`, `@aws-sdk/client-iot`, `@ts-morph/common` and `graphql` are all in the startup graph for `--version`.

## Notes for review

The tracer lives in `@webiny/project/utils/trace/` because that's the lowest package shared by both CLI entrypoints, the SDK bootstrap and the render worker.

With tracing off, `trace`/`traceAsync` return before touching the recorder. The one exception is `GetCliRunnerService`, which calls `performance.now()` twice per registered command (~50 calls, microseconds) because the phase label isn't known until the definition is built.

`formatTraceReport` has tests covering the nesting reconstruction and the filtering. `@webiny/project` had no vitest config, so this adds one along with `vitest` as a dev dependency.

Workspace: wby-next2 · ⚡ instrument slow webiny cli

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added an optional `--trace` flag for CLI commands.
  * Added timing reports for CLI startup, configuration, validation, and command execution.
  * Trace output includes phase durations, percentages, nested operations, and unfinished phases.
  * Trace information is forwarded from worker processes when enabled.

* **Tests**
  * Added coverage for trace report formatting, nested phases, hidden short phases, and incomplete operations.
  * Added test configuration for the project package.

### CODE REVIEW

**Comment by adrians5j** [comment]
On the Slop Cop finding about `packages/cli-core/files/references.json` — I traced it, and it's the expected consequence of this PR.

The file is a single-line JSON blob, so any change renders as a whole-file rewrite. Parsed and diffed semantically, here is every difference against `next`:

```
devDependencies:  vitest@4.1.11  gained  /packages/project/package.json
references:       vitest@4.1.11  gained  /packages/project/package.json
```

241 dependency entries before and after, nothing added, removed or reordered otherwise. That single addition is this PR adding `vitest` to `@webiny/project`'s devDependencies so the new `__tests__` can run, and the file is regenerated by `yarn webiny sync-dependencies`, which is a required pre-commit step in `AGENTS.md`. Not hand-edited, not stale.

Worth noting the same file content passed this check on the previous push and failed on this one, so the check is non-deterministic rather than reacting to something new. I'd rather leave it red and explain than re-push to reroll it.

File: packages/project/src/utils/trace/formatTraceReport.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+};
+
+/*
+ * A run records a lot of phases that turn out to be instant: every command definition, every
+ * extension type. Dropping the ones that round to zero and contain nothing keeps the report to the
+ * handful of rows worth reading; a hidden phase is by definition not the bottleneck.
+ */
+const isWorthPrinting = (node: ITraceNode, index: number, nodes: ITraceNode[]) => {
+    const duration = node.entry.end - node.entry.start;
+    if (Math.round(duration) > 0) {
```
[the remark this answers is not in the document]
Comment (adrians5j) [reply packages/project/src/utils/trace/formatTraceReport.ts:40]: Not taking this one, because it would invert the intent.

`formatRow` prints `Math.round(duration)`, so the filter rounds the same way on purpose: the rows kept are exactly the rows that would show a non-zero figure. Switching to `duration >= 1` would hide a 0.6 ms phase that the report renders as `1 ms`, which is the worse of the two inconsistencies — you'd have a phase suppressed while its own formatting says it was worth a millisecond.

The suggested test is a good idea in the other direction though, so I added the boundary case pinning 0.6 ms as kept and displayed as `1 ms`, plus a comment at `formatTraceReport.ts` saying why the rounding is deliberate. The existing tests already cover 0.1 ms and 0.2 ms phases being dropped.

The `startTrace.ts` finding was right and is fixed: `--trace false` now resolves to off. The parsing moved out into `isTraceRequested.ts` as a pure function over `argv`, with nine tests covering the bare flag, `=true`/`=false`, separated `true`/`false`, end-of-argv, a following flag, and a `--trace-level` near-miss.
